test/rewards/AutoCompoundingRewards.ts

```diff
@@ -686,6 +737,85 @@ describe('AutoCompoundingRewards', () => {
+                        { tokenSymbol: TokenSymbol.TKN4, initialUserStake: toWei(70_000), totalRewards: toWei(17_000) },
+                        { tokenSymbol: TokenSymbol.TKN5, initialUserStake: toWei(80_000), totalRewards: toWei(18_000) }
+                    ];
+
+                    const tokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+                    const poolTokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+
+                    beforeEach(async () => {
+                        for (const [index, setup] of setups.entries()) {
+                            ({ token: tokens[index], poolToken: poolTokens[index] } = await prepareSimplePool(
+                                new TokenData(setup.tokenSymbol),
+                                setup.initialUserStake,
+                                setup.totalRewards
+                            ));
+
+                            await createProgram(
+                                distributionType,
+                                autoCompoundingRewards,
+                                tokens[index].address,
+                                setup.totalRewards,
+                                START_TIME
+                            );
+                        }
+
+                        await autoCompoundingRewards.setAutoProcessRewardsCount(AUTO_PROCESS_REWARDS_COUNT);
+                    });
+
+                    if (distributionType === RewardsDistributionType.Flat) {
+                        it('should distribute all tokens', async () => {
+                            await autoCompoundingRewards.setTime(programEndTime[distributionType]);
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_PROCESS_REWARDS_COUNT); i++) {
+                                const res = await autoCompoundingRewards.autoProcessRewards();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                            const res = await autoCompoundingRewards.autoProcessRewards();
+                            await expect(res).not.to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                        });
+
+                        it('should distribute some tokens', async () => {
+                            for (let i = 0; i < 5; i++) {
+                                await autoCompoundingRewards.setTime(
+                                    Math.floor(START_TIME + (programDuration[distributionType] * 2 ** i) / (2 ** i + 1))
+                                );
+                                const res = await autoCompoundingRewards.autoProcessRewards();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                        });
+
+                        it('should not redistribute tokens before the minimum time has elapsed', async () => {
```

Comment:
"redistribute" == "process rewards of some pools"?

Reply:
Yes.
The original description was something like:
```
should not distribute tokens again after less than the minimum time delta for auto-processing the rewards has elapsed
```

Reply:
So please fix it

Reply:
What's wrong with the current title?

Reply:
"redistribute". The test should check the triggering - not the actual distribution (not sure why the "re" is there anyway)